include/dkm_parallel.hpp

```diff
@@ -57,16 +57,18 @@ initialization algorithm.
 */
 template <typename T, size_t N>
 std::vector<std::array<T, N>> random_plusplus_parallel(
-	const std::vector<std::array<T, N>>& data, uint32_t k, CustomGenerator& rand_engine) {
```

Comment:
Lets provide more descriptive variable names. `currentIteration` and `maxIterations` refer to the iteration of the whole kmeans. We have to keep in mind that clustering algorithm is performed more than one times, and this index is the one that we feed in here. In order not to get confused with the iteration id of each kmeans, lets use: `currentClusteringRunId`, `totalClusteringRuns` or something like that.